src/DynamoCoreWpf/ViewModels/PackageManager/PackageManagerSearchViewModel.cs

```diff
@@ -701,17 +706,27 @@ private void DownloadsOnCollectionChanged(object sender, NotifyCollectionChanged
                 foreach (var item in args.NewItems)
                 {
                     var handle = (PackageDownloadHandle)item;
-                    handle.PropertyChanged += (o, eventArgs) => this.ClearCompletedCommand.RaiseCanExecuteChanged();
+                    handle.PropertyChanged += (o, eventArgs) => ClearCompletedCommand.RaiseCanExecuteChanged();
+                    // Update CanInstall property every time the download handle's state changes
+                    handle.PropertyChanged += (o, eventArgs) => {
+                        // Hooking into propertyChanged works only if each download handle is added to
+                        // the Downloads collection before Download/Install begins.
+                        if (eventArgs.PropertyName == nameof(PackageDownloadHandle.DownloadState)) {
+                            PackageManagerSearchElementViewModel sr = SearchResults.FirstOrDefault(x => x.Model.Name == handle.Name);
+                            if (sr == null) return;
+
+                            sr.CanInstall = CanInstallPackage(handle);
+                        }
+                    };
```

Comment:
how do we unsubscribe these anonymous handlers for each handle later?

Reply:
uhm...good catch
I missed that